### ISSUE TITLE
How to set 'records_per_epoch' in the yaml.

### ISSUE THREAD

**Issue by caiduoduo12138** [determined-ai/determined#5564]
### Describe your question

My dataset has 500 images. batch_size is set to 4 per gpu. How to set the 'records_per_epoch' in the yaml. Can you give me a example? I want to run by epoch mode. Does it mean the num of iterations in one epoch.  If possible, please give more instructions.  I set  the 'records_per_epoch'  to 5000 and set epochs to '2', but it does not train with 2*5000 iterations(steps). 
config:
name: myexperiment 
bind_mounts:
     \- host_path: /mnt/
    container_path: /mnt/
    read_only: true
hyperparameters: 
  learning_rate: 0.001
  global_batch_size: 4
resources: 
  slots_per_trial: 4
scheduling_unit: 1
records_per_epoch: 498
searcher:  
  name: single
  metric: validation_loss
  max_length:
      epochs: 10
  smaller_is_better: true
entrypoint: model_def:MTrial

**Comment by rb-determined-ai** [comment]
If there's a particular number of batches you want to train for, you should configure your searcher in terms of batches.

The training loop actually counts batches, not records, and not epochs.  If you configure it in terms of records or epochs, it will just do some "dumb" math.

For example, if records_per_epoch is 5000 and epochs is 2 and global_batch_size is 4, the training loop will assume there are 5000/4=1250 batches per epoch, and so should train for 1250*2=2500 batches.

We are working on a training API where this sort of dumb math isn't necessary.  We want to support configuring training length in terms of either batches or epochs, even if your epochs are varying in length.  But today, all we really do is convert blindly to batches and count batches.

**Comment by caiduoduo12138** [comment]
Thanks for your help!
